common/xdc/nDCHistoryResender.go

```diff
@@ -126,15 +127,38 @@ func (n *NDCHistoryResenderImpl) SendSingleWorkflowHistory(
 		}
 		historyBatch := result.(*historyBatch)
 
-		// we don't need to handle continue as new here
-		// because we only care about the current run
-		// TODO: revisit to evaluate if we need to handle continue as new
 		replicationRequest := n.createReplicationRawRequest(
 			domainID,
 			workflowID,
 			runID,
 			historyBatch.rawEventBatch,
 			historyBatch.versionHistory.GetItems())
+		// This is the last batch
+		if !historyIterator.HasNext() {
+			lastEvent, err := n.getLastEvent(historyBatch.rawEventBatch)
+			if err != nil {
+				return err
+			}
+			continueAsNewAttribute := lastEvent.GetWorkflowExecutionContinuedAsNewEventAttributes()
+			if continueAsNewAttribute != nil {
+				newRunID := continueAsNewAttribute.GetNewExecutionRunId()
+				resp, err := n.getHistory(
+					domainID,
+					workflowID,
+					newRunID,
+					common.Int64Ptr(common.FirstEventID-1),
+					common.Int64Ptr(lastEvent.GetVersion()),
+					nil,
+					nil,
+					nil,
+					continueAsNewPageSize)
+				if err != nil {
+					return err
+				}
+				replicationRequest.NewRunEvents = resp.HistoryBatches[0]
+			}
```

Comment:
how about turning this into a function?

Reply:
plz also double check on the slice size `replicationRequest.NewRunEvents = resp.HistoryBatches[0]`